### ISSUE TITLE
The @codingStandardsIgnore has no effect on OneClassPerFile rule

### ISSUE THREAD

**Issue by Sphirate** [squizlabs/PHP_CodeSniffer#508]
// @codingStandardsIgnoreStart
class CPM_Exception extends Exception {}
// @codingStandardsIgnoreEnd

It does not work

**Comment by gsherwood** [comment]
The coding standard comments mute errors for lines between them. Look at the line numbers in the error output and make sure you are muting the correct ones.

For example, this file:
```php
<?php
class MyClass {}

class MySecondClass {}
```
Produces an error on line 4:  `4 | ERROR | [ ] Only one class is allowed in a file` so you need to mute the errors like this:
```php
<?php
class MyClass {}
//@codingStandardsIgnoreStart
class MySecondClass {}
//@codingStandardsIgnoreEnd
```

**Comment by Sphirate** [comment]
<?php
//@codingStandardsIgnoreStart
class MyClass {}
//@codingStandardsIgnoreEnd
class MySecondClass {}
?>

This will cause an error on line 5

**Comment by aik099** [comment]
I guess each class is reported as error in error in ignored class isn't shown, but it is affecting error detection for other class.

**Comment by Sphirate** [comment]
<?php
//@codingStandardsIgnoreStart
class MyClass {}
class MySecondClass {}
//@codingStandardsIgnoreEnd
class MyLastClass {}
?>
This will cause an error at line 6.

<?php
//@codingStandardsIgnoreStart
class MyClass {}
class MySecondClass {}

class MyLastClass 
//@codingStandardsIgnoreEnd
{
   some_code_here
}
?>
In this case, it will work correctly.

**Comment by aik099** [comment]
@Sphirate can you please use fenced code blocks, so that syntax highlight/formatting will be enabled for code pieces?

**Comment by gsherwood** [comment]
The sniff produces an error for every class it finds *after* the first class. You would need to wrap every other class declaration with ignore tags if you really want to ignore them all.

Or, just don't use this sniff in the first place if you put multiple classes in the same file.

Or, add a sniff-specific ignore pattern so tis sniff is not run over this file (or a directory of files, or a regex). Something like this:
```xml
 <rule ref="Generic.Files.OneClassPerFile">
    <exclude-pattern>*/tests/*</exclude-pattern>
 </rule>
```
